### PR TITLE
Feature/plot

### PR DESCRIPTION
**Pull request by ryujimorita**
## Change
Added plot feature. 

After the training is done, train/loss loss curve, both CER/WER metrics images are saved in `{output_dir}/images/`.

# Note
**Please install `ipykernel` in your environment by running below poetry command to avoid matplotlib error**:
`poetry run pip install ipykernel`

Screenshot 2024-12-06 at 2 11 14

### CODE REVIEW

File: src/viz.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        
+        # get only columns we use for plot
+        use_columns = [
+            "loss", "learning_rate", "epoch", "step", "eval_loss", "eval_wer", 
+            "eval_cer"
+        ]
+        df = df[use_columns].rename(columns={'loss': 'train_loss'})
+
+        # handle duplicate rows for the same step by aggregating values
+        df = df.groupby('step').mean(numeric_only=True)
```
Comment (ryujimorita) [inline src/viz.py:50]: this looks weird, but works. It's to handle the duplicates you see in the screenshot. 
it is just taking average of 1, so it is never changing the value while aligning all the mismatched rows.

Screenshot 2024-12-06 at 2 21 30

Comment (ryujimorita) [inline src/viz.py:50]: this group by average thing looks weird, but works. It's to handle the duplicates you see in the screenshot. 
it is just taking average of 1, so it is never changing the value while aligning all the mismatched rows.

Screenshot 2024-12-06 at 2 21 30
